archivist/runner.py

```diff
@@ -0,0 +1,181 @@
+"""
+Base runner class for interpreting yaml story files.
+
+"""
+
+from collections import defaultdict
+from json import dumps as json_dumps
+from logging import getLogger
+from time import sleep as time_sleep
+from typing import Dict, Optional
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+from . import archivist as type_helper
+from .errors import ArchivistError, ArchivistInvalidOperationError
+
+LOGGER = getLogger(__name__)
+
+
+def tree():
+    """Recursive dicts of dicts"""
+    return defaultdict(tree)
+
+
+class _Runner:
+    """
+    ArchivistRunner takes a url, token_file.
+    """
+
+    def __init__(self, archivist: "type_helper.Archivist"):
+        self._archivist = archivist
+        self.entities = None
+
+    def __str__(self) -> str:
+        return f"Runner({self._archivist.url})"
+
+    def __call__(self, config: Dict):
+        """
+        The dict config contains a list of `operations` to be performed serially, e.g.
+
+        ```
+        "operations": [
+            {
+                "operation": "assets.create",
+                "to_print": 10,
```

Comment:
Would be more helpful to show an example of a text prompt here, since that's what `to_print` is for.

Reply:
fixed